frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: Set<string>,
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: Set<string>;
+  updateSelectedRows: (rows: R[]) => void;
+} => {
+  const [selectedRows, setSelectedRows] = React.useState<Set<string>>();
+
+  const onSelect = React.useCallback(
+    (_event, isSelected, rowIndex, rowData) => {
+      const uniqueUIDs: Set<string> = selectedRows ? new Set([...selectedRows]) : new Set<string>();
+      if (rowIndex === -1) {
+        if (isSelected) {
+          [...visibleRows].map((uid) => uniqueUIDs.add(uid));
+        } else {
+          for (const uid of uniqueUIDs) {
+            if (visibleRows.has(uid)) {
+              uniqueUIDs.delete(uid);
+            }
+          }
+        }
+      } else if (uniqueUIDs.has(rowData?.props?.id)) {
```

Comment:
I'd just do `else` and oneliner again

```suggestion
      } else {
        isSelected ? uniqueUIDs.add(rowData?.props?.id) : uniqueUIDs.delete(rowData?.props?.id);
      }
```